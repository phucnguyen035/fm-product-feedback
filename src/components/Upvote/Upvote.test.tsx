import { fireEvent, render, screen } from '@testing-library/react';

import { Upvote } from '.';

describe('<Upvote />', () => {
  let upvote = 1;

  beforeEach(() => {
    upvote = 1;
  });

  it('should render with correct upvote value', () => {
    render(<Upvote value={upvote} />);

    expect(screen.getByText(upvote.toString())).toBeInTheDocument();
  });

  it('should have correct class for active state', () => {
    const { container } = render(<Upvote value={upvote} active={false} />);

    expect(container).toMatchSnapshot();
  });

  it('should have correct class for inactive state', () => {
    const { container } = render(<Upvote value={upvote} active />);

    expect(container).toMatchSnapshot();
  });

  it('should be disabled', () => {
    render(<Upvote value={upvote} disabled />);

    expect(screen.getByLabelText('upvote')).toBeDisabled();
  });

  it('should update upvote value on click', () => {
    render(
      <Upvote
        value={upvote}
        onClick={() => {
          upvote += 1;
        }}
      />,
    );

    fireEvent.click(screen.getByLabelText('upvote'));

    expect(upvote).toBe(2);
  });
});
