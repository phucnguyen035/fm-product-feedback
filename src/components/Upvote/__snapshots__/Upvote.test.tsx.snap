// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Upvote /> should have correct class for active state 1`] = `
<div>
  <button
    aria-label="upvote"
    class="inline-block rounded-lg px-2 py-1.5 text-body3 font-bold transition-colors text-blue bg-gray-light hover:bg-blue/50"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="w-4"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M14.707 12.707a1 1 0 01-1.414 0L10 9.414l-3.293 3.293a1 1 0 01-1.414-1.414l4-4a1 1 0 011.414 0l4 4a1 1 0 010 1.414z"
        fill-rule="evenodd"
      />
    </svg>
    <span
      class="text-navy-light"
    >
      1
    </span>
  </button>
</div>
`;

exports[`<Upvote /> should have correct class for inactive state 1`] = `
<div>
  <button
    aria-label="upvote"
    class="inline-block rounded-lg px-2 py-1.5 text-body3 font-bold transition-colors bg-blue text-white"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="w-4"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M14.707 12.707a1 1 0 01-1.414 0L10 9.414l-3.293 3.293a1 1 0 01-1.414-1.414l4-4a1 1 0 011.414 0l4 4a1 1 0 010 1.414z"
        fill-rule="evenodd"
      />
    </svg>
    <span
      class=""
    >
      1
    </span>
  </button>
</div>
`;
